core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java

```diff
@@ -433,14 +434,18 @@ public void updateUserNamePassword(Long stackId, UserNamePasswordJson userNamePa
         }
     }
 
-    public void failureReport(Long stackId, List<String> failedNodes) {
+    public void reportHealthChange(Long stackId, Set<String> failedNodes, Set<String> newHealthyNodes) {
+        if (!Sets.intersection(failedNodes, newHealthyNodes).isEmpty()) {
```

Comment:
Maybe this check can be done in periscope?

Reply:
Others can call this CB API so we need to do it here anyway.